debezium-ddl-parser/src/test/resources/oracle/examples/ddl_create.sql

```diff
@@ -17,6 +17,144 @@ CREATE TABLE "U3"."TEST_QUEUE_TABLE"(q_name            VARCHAR2(128),msgid
+-- Create External Table
+CREATE TABLE deptXTec3
+ ORGANIZATION EXTERNAL (TYPE ORACLE_DATAPUMP DEFAULT DIRECTORY def_dir1
+ ACCESS PARAMETERS (COMPRESSION ENABLED) LOCATION ('dept.dmp'));
+
+CREATE TABLE inv_part_all_xt
+    (
+      PRODUCT_ID          NUMBER(6),
+      WAREHOUSE_ID        NUMBER(3),
+      QUANTITY_ON_HAND    NUMBER(8)
+    )
+ ORGANIZATION EXTERNAL (TYPE ORACLE_DATAPUMP DEFAULT DIRECTORY def_dir1
+ ACCESS PARAMETERS (COMPRESSION ENABLED) LOCATION ('inv_p1_xt.dmp','inv_p2_xt.dmp'));
+
+CREATE TABLE emp_load
+             (first_name CHAR(15),
+              last_name CHAR(20),
+              resume CHAR(2000),
+              picture RAW (2000))
+  ORGANIZATION EXTERNAL
+  (TYPE ORACLE_LOADER
+    DEFAULT DIRECTORY ext_tab_dir
+    ACCESS PARAMETERS
+      (FIELDS (first_name VARCHARC(5,12),
+               last_name VARCHARC(2,20),
+               resume VARCHARC(4,10000),
+               picture VARRAWC(4,100000)))
+  LOCATION ('info.dat'));
+
+CREATE TABLE emp_load
+             (first_name CHAR(15),
+              last_name CHAR(20),
+              resume CHAR(2000),
+              picture RAW(2000))
+  ORGANIZATION EXTERNAL
+  (TYPE ORACLE_LOADER
+   DEFAULT DIRECTORY ext_tab_dir
+   ACCESS PARAMETERS
+     (RECORDS
+        VARIABLE 2
+        DATA IS BIG ENDIAN
+        CHARACTERSET US7ASCII
+      FIELDS (first_name VARCHAR(2,12),
+              last_name VARCHAR(2,20),
+              resume VARCHAR(4,10000),
+              picture VARRAW(4,100000)))
+    LOCATION ('info.dat'));
+
+CREATE TABLE emp_load (first_name CHAR(15),
+                      last_name CHAR(20),
+                      year_of_birth INT,
+                      phone CHAR(12),
+                      area_code CHAR(3),
+                      exchange CHAR(3),
+                      extension CHAR(4))
+  ORGANIZATION EXTERNAL
+  (TYPE ORACLE_LOADER
+   DEFAULT DIRECTORY ext_tab_dir
+   ACCESS PARAMETERS
+     (FIELDS RTRIM
+            (first_name (1:15) CHAR(15),
+             last_name (*:+20),
+             year_of_birth (36:39),
+             phone (40:52),
+             area_code (*-12: +3),
+             exchange (*+1: +3),
+             extension (*+1: +4)))
+   LOCATION ('info.dat'));
+
+CREATE TABLE emp_load (first_name CHAR(15), last_name CHAR(20), year_of_birth CHAR(4))
+  ORGANIZATION EXTERNAL (TYPE ORACLE_LOADER DEFAULT DIRECTORY ext_tab_dir
+                         ACCESS PARAMETERS (FIELDS TERMINATED BY ','
+                                            OPTIONALLY ENCLOSED BY '(' and ')'
+                                            LRTRIM)
+                         LOCATION ('info.dat'));
+
+CREATE TABLE xtab (recno varchar2(2000))
+    ORGANIZATION EXTERNAL ( TYPE ORACLE_LOADER DEFAULT DIRECTORY data_dir
+                            ACCESS PARAMETERS ( RECORDS DELIMITED BY NEWLINE
+                                                LOAD WHEN (10:13) = 0x'00000830'
+                                                PREPROCESSOR execdir:'zcat'
```

Comment:
Also, this line fails, because of `et_directory_spec`. Could check this one too?

Reply:
I think this is because `execdir` isn't a valid `regular_id` value, right?  It seems to be that `execdir` is some type of symbolic name that Oracle interprets as some special directory location and maybe its just worth adding it either to the list of non-reserved keywords for pre12c since I am certain this is valid even in Oracle 11g.